docs/production/upgrade-or-modify.md

```diff
@@ -102,6 +102,43 @@ git_repo_url = https://github.com/zulip/zulip.git
 See also our documentation on [upgrading
 docker-zulip](https://github.com/zulip/docker-zulip#upgrading-from-a-git-repository).
 
+## Updating configuration files
+
+Zulip installations often upgrade many times over their lifetime, and
+we strive to keep all configuration files backwards-compatible.
+However, our practice of leaving the `/etc/zulip/settings.py`
+unchanged during upgrades means that there may be new features which
+are not documented in that file, since it was based on a template
+provided by an earlier version of Zulip, during the initial install.
+
+After upgrading across major versions of Zulip Server, we recommend
+comparing your `/etc/zulip/settings.py` file to the current settings
+template, which can be found in
+`/home/zulip/deployments/current/zproject/prod_settings_template.py`. We
+suggest using that updated template to update
+`/etc/zulip/settings.py`:
+
+1. Ensure you have a
+   [backup](../production/export-and-import.html#backups) of your
+   `/etc/zulip/settings.py` file.
+
+1. Copy
+   `/home/zulip/deployments/current/zproject/prod_settings_template.py`
+   to `/etc/zulip/settings-new.py`.
+
+1. Go through each variable set in `/etc/zulip/settings.py`, updating
+   `/etc/zulip/settings-new.py`, wherever that setting is found in
+   there, making it match the old value in `/etc/zulip/settings.py`.
+
+   If there are settings which you cannot find documented in
+   `/etc/zulip/settings-new.py`, check the [changelog][changelog] to see
+   if they have been removed.
+
+1. Move `/etc/zulip/settings-new.py` to `/etc/zulip/settings.py`,
+   overwriting the old configuration.
```

Comment:
Especially with your new tool, I think we can reasonably recommend that they run the tool, look at the diffs, and decide either port in changes as described here or just add new entries (since we may have releases where the churn is just a new block), rather than only suggesting this approach (which is surely the only sane one for 4.0, since we redid the whitespace).

Reply:
Knowing which entries are "new" may be a little hard.  I guess the tool can diff the old template to the new template, and spit that out?  That seems like it might make the output confusing to parse.

Reply:
Yeah, and I guess especially with this update (Where we reordered things), diffing the only and new templates is probably not going to be that helpful.

Reply:
Very not-helpful.  But for many upgrades, it'll probably be reasonable.

I guess we can address that in 5.0 or whenever we have config changes that would be amenable to this. :)